File: app/src/main/java/com/moisesborges/tvaddict/showdetails/ShowDetailsView.java
package com.moisesborges.tvaddict.showdetails;

import android.support.annotation.NonNull;

import com.moisesborges.tvaddict.models.CastMember;
import com.moisesborges.tvaddict.models.Embedded;
import com.moisesborges.tvaddict.models.Season;
import com.moisesborges.tvaddict.models.Show;
import com.moisesborges.tvaddict.mvp.BaseView;

import java.util.List;

/**
 * Created by moises.anjos on 18/04/2017.
 */

interface ShowDetailsView extends BaseView{
    void setShowName(String showName);

    void setShowImage(String imageUrl);

    void setShowSummary(String summary);

    void displaySeasons(@NonNull List<Season> seasons);

    void displaySeasonsProgress(boolean isLoading);

    void displaySeasonsNotLoaded(boolean hasError);

    void setShow(@NonNull Show show);

    void navigateToEpisodes(int showId, int seasonNumber, Embedded embeddedData);

    void displaySavedShowMessage();

    void displayCastMembers(List<CastMember> castMembers);

    void displaySaveShowButton(boolean shouldDisplaySavedAction);

    void displayShowRemovedMessage();
}

File: app/src/main/java/com/moisesborges/tvaddict/showdetails/ShowDetailsView.kt
package com.moisesborges.tvaddict.showdetails

import com.moisesborges.tvaddict.models.CastMember
import com.moisesborges.tvaddict.models.Externals
import com.moisesborges.tvaddict.models.Season
import com.moisesborges.tvaddict.models.Show
import com.moisesborges.tvaddict.mvp.BaseView

/**
 * Created by moises.anjos on 18/04/2017.
 */

interface ShowDetailsView : BaseView {
    fun setShowName(showName: String)

    fun setShowImage(imageUrl: String)

    fun setShowSummary(summary: String)

    fun displaySeasons(seasons: List<Season>)

    fun displayAdditionalInfoLoadingInProgress(isLoading: Boolean)

    fun displayAdditionalInfoNotLoaded(hasError: Boolean)

    fun setShow(show: Show)

    fun navigateToEpisodes(show: Show, seasonNumber: Int)

    fun displaySavedShowMessage()

    fun displayCastMembers(castMembers: List<CastMember>)

    fun displaySaveShowButton(shouldDisplaySavedAction: Boolean)

    fun displayShowRemovedMessage()

    fun setShowRating(rating: Double?)

    fun setShowRuntime(runtime: Int?)

    fun setShowNetwork(networkName: String)

    fun setShowGenres(genres: List<String>)

    fun setShowExternalLinks(externals: Externals)
}
